### PR TITLE
Added go storage timeout, max-retry, retry-multiplier flags

### PR DESCRIPTION
**Pull request by Tulsishah**

### CODE REVIEW

File: flags.go [hunk 1]
```diff
@@ -209,6 +209,21 @@ func newApp() (app *cli.App) {
 					"inodes.",
 			},
 
+			cli.DurationFlag{
+				Name:  "Go-Storage-timeout",
```
Comment (vadlakondaswetha) [inline flags.go:213]: see the format of other flags - all letters should start with lower case

File: flags.go [hunk 2]
```diff
@@ -234,6 +249,11 @@ func newApp() (app *cli.App) {
 					"This is effective when --disable-http2 is set.",
 			},
 
+			cli.IntFlag{
+				Name:  "max-idle-conns-per-host",
+				Value: 100,
```
Comment (vadlakondaswetha) [inline flags.go:254]: did you check pranjal results. At what value we got maximum performance"

File: flags.go (hunk 1 above)
Comment (Tulsishah) [reply flags.go]: fixed

File: flags.go (hunk 2 above)
Comment (Tulsishah) [reply flags.go:254]: value is 100 only

File: flags.go [hunk 3]
```diff
@@ -209,6 +209,21 @@ func newApp() (app *cli.App) {
 					"inodes.",
 			},
 
+			cli.DurationFlag{
+				Name:  "go-storage-timeout",
```
Comment (vadlakondaswetha) [inline flags.go:213]: rename this to gcs-call-timeout and other flags just as max-retry-duration and retry-multiplier.

File: flags.go [hunk 4]
```diff
@@ -209,6 +209,21 @@ func newApp() (app *cli.App) {
...
+				Value: 800 * time.Millisecond,
+			},
+
+			cli.DurationFlag{
```
Comment (vadlakondaswetha) [inline flags.go:217]: please add usage param to all fields

Comment (Tulsishah) [reply flags.go:217]: fixed

File: flags.go (hunk 3 above)
Comment (Tulsishah) [reply flags.go]: fixed

**Comment by raj-prince** [comment]
Please write the title in sentence case. https://plos.org/resource/how-to-write-a-great-title/
Add space b/w comma and the next word.

File: flags.go [hunk 5]
```diff
@@ -234,6 +252,13 @@ func newApp() (app *cli.App) {
 					"This is effective when --disable-http2 is set.",
 			},
 
+			cli.IntFlag{
+				Name:  "max-idle-conns-per-host",
+				Value: 100,
+				Usage: "The number of MaxIdleConnsPerHost parameter in " +
```
Comment (vadlakondaswetha) [inline flags.go:258]: this comment doesn't look correct. please check.

File: flags.go [hunk 6]
```diff
@@ -344,14 +369,18 @@ type flagStorage struct {
...
-	TypeCacheTTL      time.Duration
-	LocalFileCache    bool
-	TempDir           string
-	DisableHTTP2      bool
-	MaxConnsPerHost   int
+	MaxRetrySleep       time.Duration
+	StatCacheCapacity   int
+	StatCacheTTL        time.Duration
+	TypeCacheTTL        time.Duration
+	HttpClientTimeOut   time.Duration
```
Comment (vadlakondaswetha) [inline flags.go:376]: O should be small in timeout

File: flags.go (hunk 5 above)
Comment (Tulsishah) [reply flags.go]: fixed

File: flags.go (hunk 6 above)
Comment (Tulsishah) [reply flags.go]: fixed

**Review by vadlakondaswetha — APPROVED** [review]

**Pull request merged**